examples/example-next-13/next.config.mjs

```diff
@@ -0,0 +1,64 @@
+import webpack from "webpack"
+
+const ROUTE_DECLARATION_FILE = "../types/link.d.ts"
+class TypedRoutePlugin {
+
+  apply(compiler) {
+      compiler.hooks.compilation.tap("TypedRoutePlugin", (compilation) => {
+          compilation.hooks.processAssets.tapAsync(
+              {name: "TypedRoutePlugin", stage: webpack.Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE_TRANSFER},
+             async (assets, callback) => {
+                  if (assets[ROUTE_DECLARATION_FILE]) {
+                      const original = assets[ROUTE_DECLARATION_FILE].source();
+                      const modified = original + `
+declare module "next-intl/link" {
+  import type { LinkProps as OriginalLinkProps } from "next/dist/client/link.js";
+  import type { AnchorHTMLAttributes, DetailedHTMLProps } from "react";
+  import type { UrlObject } from "url";
+
+  type LinkRestProps = Omit<
+      Omit<
+          DetailedHTMLProps<
+              AnchorHTMLAttributes<HTMLAnchorElement>,
+              HTMLAnchorElement
+          >,
+          keyof OriginalLinkProps
+      > &
+      OriginalLinkProps,
+      "href"
+  >;
+
+  export type LinkProps<RouteInferType> = LinkRestProps & {
+    /**
+     * The path or URL to navigate to. This is the only required prop. It can also be an object.
+     * @see https://nextjs.org/docs/api-reference/next/link
+     */
+    href: __next_route_internal_types__.RouteImpl<RouteInferType> | UrlObject;
```

Comment:
Oh wow, thanks for proposing this!

I haven't experimented with typed routes from Next.js yet, but based on the docs, shouldn't this be achievable by composing `Route` from `next`? See [the docs](https://nextjs.org/docs/app/building-your-application/configuring/typescript#statically-typed-links).

Btw. we do already have a webpack plugin in https://github.com/amannn/next-intl/pull/149 that creates an alias for the RSC config. If a webpack plugin is the only way to achieve this, we could integrate with that.

Reply:
Thank you for the review.
I appreciate your feedback. It seems that I missed some important details in the documentation.
At this point, it appears that using Route directly is a much better approach.
I look forward to contributing again once typedRoute stabilizes. Thank you.